### DISCUSSION TITLE
Render Priority and Coincident Topology Parameters

### DISCUSSION THREAD

**Discussion by Jimmy-KL** [pyvista/pyvista#4556] [Show and tell]
Maybe we should add coincident parameters for Pyvista to solve the flickering problem when rendering the same object at the same place. In some other software, it's called render priority.  If we want to have a similar model as follows - the ground surface is at the same location as the top surface of the geo-model, but using different scalars. In this scene, we want to give the ground surface a higher render priority.

![image](https://github.com/pyvista/pyvista/assets/114913030/59943a65-2ea4-4353-9648-9953e57a15ac)

**Comment by banesullivan** [comment]
There's a great example in https://discourse.vtk.org/t/how-to-correctly-use-the-coincidenttopology-related-methods-on-a-mapper/6888/3

But here's a simple example with `SetResolveCoincidentTopologyPolygonOffsetParameters`

```py
import pyvista as pv
from pyvista.plotting.opts import ElementType, PickerType
from pyvista import examples
import vtk

# VTK configurations to improve rendering
vtk.vtkMapper.SetResolveCoincidentTopologyToPolygonOffset()
vtk.vtkMapper.SetResolveCoincidentTopologyPolygonOffsetParameters(1.0, 1.0)

mesh = examples.load_random_hills()
cntrs, edges = mesh.contour_banded(7)

pl = pv.Plotter()
pl.add_mesh(cntrs, scalars='Scalars')
pl.add_mesh(edges, line_width=5)
pl.show()
```


| Default | (1.0, 1.0) |
|---|---|
| ![screenshot (19)](https://github.com/pyvista/pyvista/assets/22067021/358e2d87-32a4-49a7-a44a-74838cc2337a) |  ![screenshot (20)](https://github.com/pyvista/pyvista/assets/22067021/679d1193-1b5f-49a4-ab50-9fdd17e14c49) |
| ![screenshot (21)](https://github.com/pyvista/pyvista/assets/22067021/21c47762-616f-41ab-83ac-615d4261266a) |  ![screenshot (22)](https://github.com/pyvista/pyvista/assets/22067021/53653166-c91c-4122-becb-c7c85f9bab5e) |

**Reply by Jimmy-KL** [reply]
It's a bit complicated for me, but I can feel it's a great example. I will look deep into it when I have more time. Thanks!
